components/license-control-service/pkg/storage/storage.go

```diff
@@ -0,0 +1,248 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/lib/pq"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/chef/automate/components/license-control-service/pkg/keys"
+	"github.com/chef/automate/lib/db/migrator"
+	"github.com/chef/automate/lib/logger"
+)
+
+// NoLicenseError is the error returned by a backend when it does not
+// have a configured License.
+type NoLicenseError struct{ backend string }
+
+func (n *NoLicenseError) Error() string {
+	return fmt.Sprintf("No license found in %s storage backend", n.backend)
+}
+
+// RetriableBackendError is the error returned by a backend when an
+// update has failed but may work if retried.
+type RetriableBackendError struct{ err error }
```

Comment:
For some reason I thought we had a linter that would explode unless these were named in the `ErrRetriableBackend` style. Guess not!

Reply:
👉 `FooError` for types, `ErrFoo` for values: https://github.com/golang/go/wiki/Errors#naming

I _wish_ we had a linter for that 😃